### ISSUE TITLE
allow subdomains

### ISSUE THREAD

**Issue by exmatrikulator** [alexalouit/ISPConfig-letsencrypt#18]
allow subdomains, which are configured in ispconfig.

**Comment by JustDevZero** [comment]
As far as I've seen "Let's encrypt" in that patch is asking for a domain/subdomain and It's respective www", so... I just thit the trick to create "www.mysubdomain.mydomain.tld" after creating "subdomain.mydomain.tld" and It worked.

I haven't check if there is some way to avoid that "www" need, but for know worked for me.

**Comment by exmatrikulator** [comment]
it's ignoring the ispconfig entries under subdomains :(

**Comment by Blicelm** [comment]
Any news for subdomain ?

**Comment by JustDevZero** [comment]
Instead of creating them as "subdomains", try to create they as normal "Websites", and in the "Auto-Subdomain" instead of selecting "www", select "none", this should work.

**Comment by Blicelm** [comment]
Can you add an option for vhost subdomain ? I don't show let's encrypt on it.
I can create certificate for it but not with let's encrypt.

**Comment by alexalouit** [comment]
You're right. Let me some days to see that.

**Comment by mattanja** [comment]
+1 subdomain config would come in handy. Will your plugin/ISPConfig overwrite the settings file in /etc/letsencrypt/renewal/*.conf if I add sub-domains manually?

Because that would probably be the easiest workaround: first let this plugin setup the domain and the symlink, then update the certificate by executing:

```
/opt/letsencrypt/letsencrypt-auto certonly \
-w /var/www/clients/client111/web111/web \
-d your-domain.xy \
-d a.your-domain.xy \
-d b.your-domain.xy \
-d c.your-domain.xy \
-d d.your-domain.xy \
-d e.your-domain.xy \
--expand
```

You'll also need to update the renawal config in /etc/letsencrypt/renewal/your-domain.xy.conf and update the list of domains.

(And btw: thanks for this great plugin, worked like a charm out of the box!)

**Comment by alexalouit** [comment]
fix in f74f4a1
